scrapy/contrib/exporter/__init__.py

```diff
@@ -154,6 +154,7 @@ def _export_xml_field(self, name, serialized_value):
     if sys.version_info[:3] >= (2, 7, 4):
         def _xg_characters(self, serialized_value):
             if not isinstance(serialized_value, unicode):
+                serialized_value = str(serialized_value)
```

Comment:
If it is a non-ascii byte string it will be decoded to unicode using self.encoding. `str(serialized_value)` will do nothing in this case. If the string is unicode `str(serialized_value)` won't be called (see the condition above).